### PR TITLE
Usage markdown doc generation

### PR DESCRIPTION
**Pull request by fredbi**
* added a "usage" command that generates markdown docs from cobra help
* added pre-commit (client side) githook to build datamon and generate updated doc
* explained githook config in contributing guidelines

TODO:
* [ ] to be defined: do we want usage doc to be versioned, and possibly generated only on tagged release.
* [ ] to be defined: we may generate man pages and bundle this with released artifacts
Signed-off-by: Frederic BIDON <frederic@oneconcern.com>

### CODE REVIEW

File: cmd/datamon/cmd/bundle_list.go [hunk 1]
```diff
@@ -27,7 +27,12 @@ func applyBundleTemplate(bundle model.BundleDescriptor) error {
 var BundleListCommand = &cobra.Command{
 	Use:   "list",
 	Short: "List bundles",
-	Long:  "List the bundles in a repo, ordered by their bundle ID",
+	Long: `List the bundles in a repo, ordered by their bundle ID.
+
+This is analogous to the "git log" command. The bundle ID works like a git commit hash.`,
+	Example: `% datamon bundle list --repo ritesh-test-repo
+Using config file: /Users/ritesh/.datamon/datamon.yaml
```
Comment (kerneltime) [inline cmd/datamon/cmd/bundle_list.go]: Is the config file output still printed?

**Comment by fredbi** [comment]
@ritesh Ok will go through examples to accommodate new changes

File: cmd/datamon/cmd/bundle_list.go (hunk 1 above)
Comment (fredbi) [reply cmd/datamon/cmd/bundle_list.go]: fixed this (reviewed through all examples)

**Comment by fredbi** [comment]
fixed this (reviewed through all examples)

**Pull request merged**